src/main_functions.c

```diff
@@ -0,0 +1,530 @@
+	underline[0] = '\0'; 
+
+	while(misspelled[element] != NULL) {
+
+		char *ptr = strstr(sentence, misspelled[element]);
+
+		if(ptr != NULL) {
+
+			int pos = ptr - sentence;
+
+			int i = 0;
+			size_t j = 0;
+
+			for(; i < pos; i++) {
+
+				strcat(underline, " ");
+
+			}
+
+			for (; j < strlen(misspelled[element]); j++) {
+
+				strcat(underline, "^");
+
+			}
+
+			sentence = ptr+strlen(misspelled[element]);
+
+		}
+
+		element++;
+
+	}
+
+	return underline;
+
+}
+
+
+
+int add_to_misspelled(char *word, char** misspelled)
+{
+	if (word == NULL || misspelled == NULL) {
+		return EXIT_FAILURE;
+	}
+	int i = 0;
+	while(misspelled[i] != NULL) {
+		i++;
+	}
+	misspelled[i] = word;
+	return EXIT_SUCCESS;
+	// printf("adding badword %s", badword);
+	// printf("i is %d ", i);
+}
+
+int parse_string(char* string, dict_t *dict, char *underline, char** misspelled)
+{
+	char *tkn = strtok(string, ": ,.-'\n'" ""); //words only separated by these punctuation
+	while (tkn != NULL) {
+
+		if (valid_word(tkn, dict) == EXIT_FAILURE){
+			//printf("%s ", tkn);
+			underline_misspelled(tkn, underline);
+			add_to_misspelled(tkn, misspelled);
+		}
+		else if (valid_word(tkn, dict) == EXIT_SUCCESS) {
+			//printf("%s ", tkn);
+			underline_correct_spelling(tkn, underline);
+		}
+		else {
+			printf("error processing text");
+			return EXIT_FAILURE;
+		}
+		tkn = strtok(NULL, " ,.-");
+	}
+	return EXIT_SUCCESS;
+
+}
+
+//reference from https://stackoverflow.com/questions/32413667/replace-all-occurrences-of-a-substring-in-a-string-in-c
+char* correct_line(char* line, char* old_word, char* new_word)
+{
+	char buffer[2000] = {0}; //again, we might need to modify our size estimates
+	char *insert_point = &buffer[0];
+	char *tmp = line;
+	int old_length = strlen(old_word);
+	int new_length = strlen(new_word);
+
+	while(1) {
+		char *p = strstr(tmp, old_word);
+
+		if (p == NULL) {
+			strcpy(insert_point, tmp);
+			break;
+		}
+
+		memcpy(insert_point, tmp, p - tmp);
+		insert_point += p - tmp;
+
+		memcpy(insert_point, new_word, new_length);
+		insert_point += new_length;
+
+		tmp = p + old_length;
+	}
+
+	strcpy(line, buffer);
+
+	return line;
+
+}
+
+
+
+int initialize_misspelled(char **misspelled, int length)
+{
+
+	if (misspelled==NULL) {
+		return EXIT_FAILURE;
+	}
+    int i = 0;
+	for ( ; i < length; i++) {
+    	misspelled[i] = NULL; //initialize each element to be NULL
+    }
+
+    return EXIT_SUCCESS;
+}
+
+
+
+/*
+ *	III. Interactive Mode
+ */
+
+char* edit_interactive(char* line, dict_t* dict, int linenumber)
+{
+    char *line_copy = malloc(strlen(line));
+    strcpy(line_copy, line); //maintain a copy of the line to preserve original line: line will be parsed into individual words
+    
+    char *line_copy2 = malloc(strlen(line));
+    strcpy(line_copy2, line); 
+    int max_no_suggestions = 2; //should the user decide this?
+
+
+    int length = strlen(line) / 3; //approximate 3 chars per word to be safe
```

Comment:
**WOAH STOP** why is this happening? That's not emperical in the slightest. Why would we assume anything about the strings the user is trying to input?

Reply:
updated to just strlen(line)